adoc/admin-gpus.adoc

```diff
@@ -0,0 +1,182 @@
+
+OCI hooks are a way for vendors or projects to inject executable actions into the lifecycle of a container managed by the container runtime (runc). SUSE provides an OCI hook for NVIDIA that enable the container runtime and therefor the kubelet and the Kubernetes scheduler to query the host system for the presence of a GPU device and access it directly. Install the hook on the worker nodes with GPUs:
+
+[source,bash]
+----
+# zypper install nvidia-container-toolkit
+----
+
+=== Test a GPU container image
+
+At this point, you should be able to run a container image that requires a GPU and directly access the device from the running container, for example using podman:
+
+[source,bash]
+----
+# podman run docker.io/nvidia/cuda nvidia-smi
+----
+
+=== Troubleshooting
+
+At this point, you should be able to run a container image using a GPU. If that is not working, check the following:
+
+Ensure your GPU is visible from the host system:
+
+[source,bash]
+----
+# lspci | grep -i nvidia
+# nvidia-smi
+----
+
+Ensure the kernel modules are loaded:
+
+[source,bash]
+----
+# lsmod | grep nvidia
+----
+
+If they are not, try loading them explicitly and check dmesg for an error indicating why they are missing:
+
+[source,bash]
+----
+# nvidia-modprobe
+# dmesg | tail
+----
+
+== Configure Kubernetes
+
+=== Install the device plugin
+
+The Kubernetes device plugin framework allows the kubelet to advertise system hardware resources that the Kubernetes scheduler can then use as hints to schedule workloads that require such devices. The Kubernetes device plugin from NVIDIA allows the kubelet to advertise NVIDIA GPUs it finds present on the worker node. Install the device plugin using kubectl:
+
+[source,bash]
+----
+$ kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/1.0.0-beta6/nvidia-device-plugin.yml
+----
+
+=== Taint GPU workers
+
+In a heterogeneous cluster, it may be preferable to prevent scheduling pods that do not require a GPU on nodes with a GPU in order to ensure that GPU workloads are not competing for time on the hardware they need to run. To accomplish this, add a taint to the worker nodes that have GPUs:
+
+[source,bash]
+----
+$ kubectl taint nodes worker0 nvidia.com/gpu=:NoSchedule
+----
+
+or
+
+[source,bash]
+----
+$ kubectl taint nodes worker0 nvidia.com/gpu=:PreferNoSchedule
+----
+
+See the Kubernetes documentation on link:{kubedoc}concepts/scheduling-eviction/taint-and-toleration/[taints and tolerations] for a discussion on the considerations for using the NoSchedule or PreferNoSchedule effects.
+
+The ExtendedResourceToleration admission controller will automatically add the nvidia.com/gpu toleration to pods that request the nvidia.com/gpu resource, so you will not need to add this toleration explicitly for every GPU workload. When after initializing the cluster but before bootstrapping it, add the ExtendedResourceToleration to the list of enabled admission plugins in your cluster's kubeadm-init.conf file:
```

Comment:
`The link:{kubedoc}reference/access-authn-authz/admission-controllers/#extendedresourcetoleration[ExtendedResourceToleration admission controller]`

Reply:
Done